smartReceiptsLibrary/src/main/java/co/smartreceipts/android/rating/data/AppRatingPreferencesStorage.java

```
@@ -0,0 +1,116 @@
+package co.smartreceipts.android.rating.data;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import java.util.concurrent.Callable;
+
+import rx.Single;
+
+public class AppRatingPreferencesStorage implements AppRatingStorage {
+
+    private Context mAppContext;
+
+    private static final class Keys {
+        /**
+         * Key to get rating preferences
+         */
+        private static final String RATING_PREFERENCES = "Smart Receipts rating";
+        /**
+         * Key to track user preference about no longer showing rating window
+         */
+        private static final String DONT_SHOW = "dont_show";
+
+        /**
+         * Key to track how many times the user has launched the application
+         */
+        private static final String LAUNCH_COUNT = "launches";
+
+        /**
+         * Key to track if the users wishes to be reminded later
+         */
+        private static final String ADDITIONAL_LAUNCH_THRESHOLD = "threshold";
+
+        /**
+         * Key to track the first call of {@link AppRatingStorage#incrementLaunchCount()} method in millis
+         */
+        private static final String INSTALL_TIME_MILLIS = "days";
+
+        /**
+         * Key to track if the application crashed at a prior date
+         */
+        private static final String CRASH_OCCURRED = "hide_on_crash";
+    }
+
+    public AppRatingPreferencesStorage(Context context) {
```

@NonNull

And Preconditions.checkNotNull(context.getAppContext)

Helps us to fail fast